bigquery/google/cloud/bigquery/client.py

```diff
@@ -239,18 +267,26 @@ def create_table(self, table):
             method='POST', path=path, data=resource)
         return Table.from_api_repr(api_response, self)
 
-    def get_dataset(self, dataset_ref):
+    def get_dataset(self, dataset_ref, retry=DEFAULT_RETRY):
         """Fetch the dataset referenced by ``dataset_ref``
 
         :type dataset_ref:
             :class:`google.cloud.bigquery.dataset.DatasetReference`
         :param dataset_ref: the dataset to use.
 
+        :type retry: :class:`google.api.core.retry.Retry`
+        :param retry: (Optional) How to retry the RPC.
+
         :rtype: :class:`google.cloud.bigquery.dataset.Dataset`
         :returns: a ``Dataset`` instance
         """
-        api_response = self._connection.api_request(
-            method='GET', path=dataset_ref.path)
+        api_call = functools.partial(
+            self._connection.api_request,
```

Comment:
I think this may be failing because `api_request` is not getting the `self` argument. Try adding `self._connection,` as a parameter.

Reply:
Nevermind, looks like it's more subtle than that. https://stackoverflow.com/a/28752007/101923

Reply:
I don't know Python well enough to translate that SO page into a fix. @lukesneeringer @jonparrott can you help?

Reply:
Ugh, wraps is such a pain. I will send a fix for this to core, but I don't have time right now as I'm fire drilling. I'll see what I can do by the end of this week.

Reply:
Created https://github.com/GoogleCloudPlatform/google-cloud-python/pull/4166

Reply:
Okay, this is now in core, so if you rebase it should work fine.